Mage/src/main/java/mage/abilities/common/DealCombatDamageControlledTriggeredAbility.java

```diff
@@ -60,34 +66,34 @@ public DealCombatDamageControlledTriggeredAbility copy() {
 
     @Override
     public boolean checkEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.DAMAGED_PLAYER
-                || event.getType() == GameEvent.EventType.COMBAT_DAMAGE_STEP_PRIORITY
-                || event.getType() == GameEvent.EventType.ZONE_CHANGE;
+        return event.getType() == GameEvent.EventType.DAMAGED_PLAYER_BATCH_ONE_PLAYER;
     }
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        if (event.getType() == GameEvent.EventType.COMBAT_DAMAGE_STEP_PRIORITY ||
-                (event.getType() == GameEvent.EventType.ZONE_CHANGE && event.getTargetId().equals(getSourceId()))) {
-            damagedPlayerIds.clear();
+        
+        List<DamagedEvent> events = ((DamagedBatchEvent) event).getEvents().stream()
+            .filter(DamagedEvent::isCombatDamage)
+            .filter(e -> {
+                Permanent permanent = game.getPermanentOrLKIBattlefield(e.getSourceId());
+                return permanent != null
+                        && filter.match(permanent, game)
+                        && permanent.isControlledBy(this.getControllerId());
+            })
+            .collect(Collectors.toList());
+        
+        if (events.isEmpty()) {
             return false;
         }
-        if (event.getType() != EventType.DAMAGED_PLAYER) {
-            return false;
-        }
-        DamagedPlayerEvent damageEvent = (DamagedPlayerEvent) event;
-        Permanent p = game.getPermanent(event.getSourceId());
-        if (!damageEvent.isCombatDamage()
-                || p == null
-                || !p.isControlledBy(this.getControllerId())
-                || damagedPlayerIds.contains(event.getPlayerId())
-                || (onlyOpponents && !game.getOpponents(getControllerId()).contains(event.getPlayerId()))) {
-            return false;
-        }
-        damagedPlayerIds.add(event.getPlayerId());
-        if (setTargetPointer) {
-            this.getEffects().setTargetPointer(new FixedTarget(event.getPlayerId()));
+
+        switch (setTargetPointer) {
+            case PLAYER:
+                this.getEffects().setTargetPointer(new FixedTarget(event.getPlayerId()));
```

Comment:
is it possible to get the total damage dealt from the events? if so, could store it as saved damage value in case future cards use that. not required though